Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -47,40 +47,68 @@
 FOUNDATION_EXPORT NSString* const NSUserDefaultsDidChangeNotification = @"NSUserDefaultsDidChangeNotification";
 
 @implementation NSUserDefaults {
+    StrongId<NSMutableDictionary> _cacheDict;
+    std::mutex _cacheLock;
+    BOOL _cacheIsDirty;
     StrongId<NSMutableDictionary> _registrationDict;
+    StrongId<NSOperationQueue> _synchronizeQueue;
 }
 
+static NSUserDefaults* _standard;
+
 /**
  @Status Interoperable
 */
 - (instancetype)init {
-    [super init];
+    return [self initWithSuiteName:nil];
+}
+
+/**
+ @Status Caveat
+ @Notes supports nil only for suitename
+*/
+- (instancetype)initWithSuiteName:(NSString*)suitename {
+    if (nil != suitename) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    if(self = [super init]) {
 
-    _registrationDict = [NSMutableDictionary dictionary];
+        _cacheDict = [NSMutableDictionary dictionary];
+        _registrationDict = [NSMutableDictionary dictionary];
 
-    [self setObject:[NSArray arrayWithObject:@"en"] forKey:@"AppleLanguages"];
-    [self setObject:@"en_US" forKey:@"AppleLocale"];
+        _synchronizeQueue.attach([NSOperationQueue new]);
+        [_synchronizeQueue setMaxConcurrentOperationCount:1];
+
+        [self setObject:[NSArray arrayWithObject:@"en"] forKey:@"AppleLanguages"];
+        [self setObject:@"en_US" forKey:@"AppleLocale"];
+    }
 
     return self;
 }
 
+- (void)dealloc {
+    [_synchronizeQueue waitUntilAllOperationsAreFinished];
+    [super dealloc];
+}
+
 /**
  @Status Interoperable
 */
 + (NSUserDefaults*)standardUserDefaults {
-    static NSUserDefaults* standard;
-
-    if (standard == nil) {
-        standard = [self new];
+    if (_standard == nil) {
```

Comment:
I think you should keep this as-is; use a magic static to force threadsafe init.
```
static NSUserDefaults* standard = [self new];
return standard;
```

C++11 ensures it only happens once.

Ideally, the suspend/resume methods would be per-instance; UIApplication would then call `[[NSUserDefaults standardUserDefaults] _zhuLiDoTheThing]`.